hera_qm/firstcal_metrics.py

```diff
@@ -0,0 +1,533 @@
+"""
+FirstCal metrics
+
+"""
+import numpy as np
+import os
+from pyuvdata import UVData, UVCal
+import hera_cal as hc
+import matplotlib
+import matplotlib.pyplot as plt
+import astropy.stats as astats
+from collections import OrderedDict
+import json
+import cPickle as pkl
+
+
+class FirstCal_Metrics(object):
+    """
+    FirstCal_Metrics class for holding firstcal data,
+    running metrics, and plotting delay solutions
+
+    FirstCal_Metrics currently only works on a 
+    per-file basis, as pyuvadata.UVCal also only
+    works on a per-file basis. It also only works
+    with files that have more than a few time integrations,
+    otherwise the standard deviation metric is not well-defined.
+    """
+
+    def __init__(self, calfits_file):
+        """
+        Input:
+        ------
+        calfits_file : str
+            filename for a *.first.calfits file
+
+        Result:
+        -------
+        self.UVC : pyuvdata.UVCal() instance
+
+        self.delsys : ndarray, shape=(N_ant, N_times)
+            firstcal delay solutions
+
+        self.delay_offsets : ndarray, shape=(N_ant, N_times)
+            firstcal delay solution offsets from time average
+
+        self.frac_JD : ndarray, shape=(N_times,)
+            ndarray containing time-stamps of each integration
+            in fraction of current JD
```

Comment:
`frac_JD` means remainder of JD after subtracting the integer value. I'm open to whatever is the clearest, so let me know if you have any ideas